### ISSUE TITLE
TypeError: can't concat int to bytes

### ISSUE THREAD

**Issue by mayomacam** [Gallopsled/pwntools#1747]
I have start learning pwntools as i was writing my program and try to disassemble elf i got this error. I try to edit and convert it to same value after some try it work starting to got error in another script. Anyway i can't make it work.
```
Traceback (most recent call last):
File "/home/mayomacam/Music/reverse-engineering/pwntool/1.py", line 31, in
print(b.disasm(b.data, 16))
File "/home/mayomacam/.local/lib/python3.9/site-packages/pwnlib/elf/elf.py", line 1489, in disasm
return disasm(self.read(address, n_bytes), vma=address, arch=arch, endian=self.endian)
File "/home/mayomacam/.local/lib/python3.9/site-packages/pwnlib/elf/elf.py", line 1379, in read
stop = address + count
TypeError: can't concat int to bytes
```
here is my program ,
```
#!/usr/bin/python3
from pwn import *
from capstone import *
context.log_level = 'debug'
_ = '='*135
_d_ = input("Where file is file loated: \n1. same directory\n2. on different location\n$ ")
print(_)
if int(_d_) == 1:
    file = input("Enter filename to check: ").replace("\n","")
    file = "./"+file
if int(_d_) ==2 :
    file = input("Enter filename to check: ").replace("\n","")
print("\n"+_)
target = process(file)   #define or attaching a elf file
print('receving data...')
print('sending data')
#fl = bytes(input())
target.sendline(b'12443658760')
#target.interactive()
print('data sent..')
target.recvall()
target.close()
print(_)
#start analyse
print(os.system('file {}'.format(file)))
print(_)
b = ELF(file)
print("starting address: {}".format(b.address))
print("address offset: {}\n".format(b.address_offsets))
print(b.disasm(b.data, 16))
```
And here is my output ,
```
Where file is file loated: 
1. same directory
2. on different location
$ 1
=======================================================================================================================================
Enter filename to check: beleaf

=======================================================================================================================================
[+] Starting local process './beleaf' argv=[b'./beleaf'] : pid 104364
receving data...
sending data
[DEBUG] Sent 0xc bytes:
    b'12443658760\n'
data sent..
[+] Receiving all data: Done (30B)
[*] Process './beleaf' stopped with exit code 1 (pid 104364)
[DEBUG] Received 0x1e bytes:
    b'Enter the flag\n'
    b'>>> Incorrect!\n'
=======================================================================================================================================
./beleaf: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 3.2.0, BuildID[sha1]=6d305eed7c9bebbaa60b67403a6c6f2b36de3ca4, stripped
0
=======================================================================================================================================
[DEBUG] PLT 0x680 puts
[DEBUG] PLT 0x690 strlen
[DEBUG] PLT 0x6a0 __stack_chk_fail
[DEBUG] PLT 0x6b0 printf
[DEBUG] PLT 0x6c0 __isoc99_scanf
[DEBUG] PLT 0x6d0 exit
[DEBUG] PLT 0x6e0 __cxa_finalize
[*] '/home/mayomacam/Music/reverse-engineering/pwntool/beleaf'
    Arch:     amd64-64-little
    RELRO:    Full RELRO
    Stack:    Canary found
    NX:       NX enabled
    PIE:      PIE enabled
starting address: 0
address offset: <bound method ELFFile.address_offsets of ELF('/home/mayomacam/Music/reverse-engineering/pwntool/beleaf')>

Traceback (most recent call last):
  File "/home/mayomacam/Music/reverse-engineering/pwntool/elf_detail.py", line 116, in <module>
    print(b.disasm(b.data, 16))
  File "/home/mayomacam/.local/lib/python3.9/site-packages/pwnlib/elf/elf.py", line 1489, in disasm
    return disasm(self.read(address, n_bytes), vma=address, arch=arch, endian=self.endian)
  File "/home/mayomacam/.local/lib/python3.9/site-packages/pwnlib/elf/elf.py", line 1379, in read
    stop = address + count
TypeError: can't concat int to bytes
```

And pwntools version details,
```
Name: pwntools
Version: 4.3.1
Summary: Pwntools CTF framework and exploit development library.
Home-page: https://pwntools.com
Author: Gallopsled et al.
Author-email: pwntools-users@googlegroups.com
License: Mostly MIT, some GPL/BSD, see LICENSE-pwntools.txt
Location: /home/mayomacam/.local/lib/python3.9/site-packages
Requires: psutil, pysocks, pip, pyelftools, pyserial, sortedcontainers, requests, capstone, mako, intervaltree, paramiko, packaging, ropgadget, pygments, unicorn, six, python-dateutil
Required-by: pwn
```

**Comment by Arusekk** [comment]
I now had a closer look that this is somewhat intended, so closing in favor of #1743.
